html/login/controllers/authentication.php

```diff
@@ -0,0 +1,48 @@
+<?php
+require_once "../models/user.php";
+
+class AuthenticationController {
+    public function authenticate($username, $password) {
+        $userModel = new UserModel();
+        $user = $userModel->authenticateUser($username, $password);
+
+        if ($user) {
+            header("Location: ../views/dashboard.php");
+            exit();
+        } else {
+            // Redirect to the login page with an error message
+            header("Location: ../views/login.php?error=Incorrect Username or Password");
+            exit();
+        }
+    }
+
+    public function handleLoginRequest() {
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            try {
+                if (!isset($_POST['username']) || !isset($_POST['password'])) {
+                    throw new Exception("Username and password are required.");
+                }
+
+                $username = $_POST['username'];
+                $password = $_POST['password'];
+
+                if (empty($username)) {
+                    throw new Exception("Username is required");
+                } elseif (empty($password)) {
```

Comment:
can you separate the condition of this.

Reply:
Yes sir, i already resolved and pushed it. Thank you.